### DISCUSSION TITLE
Update HotChocolate monorepo configuration

### DISCUSSION THREAD

**Discussion by EricStG** [renovatebot/renovate#22483] [Suggest an Idea]
The HotChocolate monorepo was enabled a while back (see #9687)

But the Github repository changed to https://github.com/ChilliCream/graphql-platform/
Additionally, there are other projects in the same repository, so the source would have to be limited HotChocolate.* packages

The HotChocolate prefix is reserved to their organization on Nuget, if that makes things easier.

**Comment by rarkins** [comment]
If a repo changes its name, it's a feature request for Renovate, not a bug report. PRs welcome

**Reply by EricStG** [reply]
It changed its name, but they also merged a bunch of different products in the same repo
I'll gladly open a PR, but I don't know what's the approach in this case

**Reply by HonkingGoose** [reply]
I think this is what we usually do:

- a new [Monorepo preset](https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/monorepo.ts) for the `ChiliCream/graphql-platform` 
- a new [Replacement preset](https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/replacements.ts) that lists the old monorepo name, and points to the newer name, that way users who have `configMigration` enabled get a update PR from Renovate to point to the new monorepo 😄 

It's been a while since I created a preset, so I may be wrong with the steps above. Wait for a maintainer to confirm/deny this approach before starting. 😉

**Reply by secustor** [reply]
The first one definitely, the second only if references in the package file have changed too.

**Comment by VILLAN3LL3** [comment]
What is the current status here? I would create a PR, but I have not yet understood how to restrict packages within a repo

**Reply by secustor** [reply]
I am not aware that anyone is working on this.  
To implement this you have to replace the [RepoGroup](https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/monorepo.ts#L6) with a [packagePattern](https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/monorepo.ts#L286)

**Reply by VILLAN3LL3** [reply]
https://github.com/renovatebot/renovate/pull/23073
